skbio/metadata/tests/test_intersection.py

```diff
@@ -0,0 +1,219 @@
+            self.assertTrue(d.start < 59)
+
+    def test_n(self):
+        iv = self.intervals
+        for i in range(0, 90, 10):
+            r = iv.after(i, max_dist=20, num_intervals=2)
+            self.assertEqual(r[0].start, i + 10)
+            self.assertEqual(r[1].start, i + 20)
+
+            r = iv.after_interval(IntervalObj(i, i),
+                                  max_dist=20, num_intervals=2)
+            self.assertEqual(r[0].start, i + 10)
+            self.assertEqual(r[1].start, i + 20)
+
+
+class LotsaTestCase(unittest.TestCase):
+    """ put lotsa data in the tree and make sure it works"""
+    def setUp(self):
+        iv = IntervalNode(1, 2, IntervalObj(1, 2))
+        self.max = 1000000
+        for i in range(0, self.max, 10):
+            f = IntervalObj(i, i)
+            iv = iv.insert(f.start, f.end, f)
+
+        for i in range(600):
+            iv = iv.insert(0, 1, IntervalObj(0, 1))
+        self.intervals = iv
+
+    def test_count(self):
+        iv = self.intervals
+
+        r = iv.right(1, n=33)
+        self.assertEqual(len(r), 33)
+
+        l = iv.left(1, n=33)
+        self.assertEqual(len(l), 1)
+
+        u = iv.right(1, n=9999)
+        self.assertEqual(len(u), 250)
+
+        # now increase max_dist
+        u = iv.right(1, n=9999, max_dist=99999)
+        self.assertEqual(len(u), 9999)
+
+    def test_max_dist(self):
+        iv = self.intervals
+        r = iv.right(1, max_dist=0, n=10)
+        self.assertEqual(len(r), 0)
+
+        for n, d in enumerate(range(10, 1000, 10)):
+            r = iv.right(1, max_dist=d, n=10000)
+            self.assertEqual(len(r), n + 1)
+
+    def test_find(self):
+        iv = self.intervals
+        path = sys.path[:]
+        sys.path = sys.path[2:]
+        random = __import__("random")
+        sys.path = path
+        for t in range(25):
+            start = random.randint(0, self.max - 10000)
+            end = start + random.randint(100, 10000)
+            results = iv.find(start, end)
+            for feat in results:
+                self.assertTrue(
+                        (feat.end >= start and feat.end <= end) or
+                        (feat.start <= end and feat.start >= start)
+                        )
+
+
+class IntervalTreeTest(unittest.TestCase):
+    def setUp(self):
+        iv = IntervalTree()
+        n = 0
+        for i in range(1, 1000, 80):
+            iv.insert(i, i + 10, dict(value=i*i))
+            # add is synonym for insert.
+            iv.add(i + 20, i + 30, dict(astr=str(i*i)))
+
+            # or insert/add an interval object with start, end attrs.
+            iv.insert_interval(IntervalObj(i + 40, i + 50,
+                                           value=dict(astr=str(i*i))))
+            iv.add_interval(IntervalObj(i + 60, i + 70,
+                                        value=dict(astr=str(i*i))))
+
+            n += 4
+        self.intervals = self.iv = iv
+        self.nintervals = n
+
+    def test_find(self):
```

Comment:
This may make refactors in the future more difficult, but since there are subsequent tests, it should be fine. Would you consider adding in a comment to this test as well as others indicating that further implicit tests are performed in `test_interval`?